components/FilterLeadsCard/FilterLeadsCard.tsx

```diff
@@ -0,0 +1,236 @@
+'use-client';
+
+import { LEADS_DATA } from '@/utils/constants/leadslist-constant';
+import { useState } from 'react';
+import { CustomMultiSelect } from '../CustomMultiSelect/custom-multi-select';
+import 'react-dates/initialize';
+import 'react-dates/lib/css/_datepicker.css';
+import { DateRangePicker } from 'react-dates';
+import { Moment } from 'moment';
+
+interface statusState {
+  hot: boolean;
+  warm: boolean;
+  cold: boolean;
+}
+
+interface FilterLeadsCardProps {
+  setFilterCardOpen: React.Dispatch<React.SetStateAction<boolean>>;
+  onFilterData: (data: any) => void;
+}
+
+const FilterLeadsCard: React.FC<FilterLeadsCardProps> = ({
+  onFilterData,
+  setFilterCardOpen,
+}) => {
+  //! State for date range
+  const [startDate, setStartDate] = useState<Moment | null>(null);
+  const [endDate, setEndDate] = useState<Moment | null>(null);
+  const [focusedInput, setFocusedInput] = useState<'startDate' | 'endDate' | null>(null);
+
+  //! State for status
+  const [status, setStatus] = useState<statusState>({
+    hot: false,
+    warm: false,
+    cold: false,
+  });
+
+  const handleCheckboxChange = (statusName: keyof statusState) => {
```

Comment:
Move this handler where all handlers are appeared, we should not keep handlers inside state declarations.